src/main/java/org/opensearch/knn/index/codec/KNN80Codec/KNN80DocValuesConsumer.java

```diff
@@ -227,6 +253,18 @@ private void createKNNIndexFromScratch(FieldInfo fieldInfo, KNNCodecUtil.Pair pa
             );
         }
 
+        // Update index description of Faiss for binary data type
+        if (KNNEngine.FAISS == knnEngine
```

Comment:
It's very hard to understand multiple conditions , Can we do the following 
1. we are calling VectorDataType.BINARY.getValue() multiple time in if condition , can we  call once. 

or probably something like this to make more readable

```
String vectorDataType = fieldAttributes.getOrDefault(KNNConstants.VECTOR_DATA_TYPE_FIELD, VectorDataType.FLOAT.getValue());
Object indexDescription = parameters.get(KNNConstants.INDEX_DESCRIPTION_PARAMETER);

if (KNNEngine.FAISS == knnEngine && VectorDataType.BINARY.getValue().equals(vectorDataType) && indexDescription != null) {
    String updatedIndexDescription = FAISS_BINARY_INDEX_DESCRIPTION_PREFIX + indexDescription.toString();
    parameters.put(KNNConstants.INDEX_DESCRIPTION_PARAMETER, updatedIndexDescription);
    parameters.put(KNNConstants.VECTOR_DATA_TYPE_FIELD, VectorDataType.BINARY.getValue());
}
```

Reply:
@Vikasht34 Thanks for the reviewing, but this part code change actually came from https://github.com/opensearch-project/k-NN/pull/1784, which is merged. As I mentioned in this PR description, I rebased [junqiu-lei:binary-ivf](https://github.com/junqiu-lei/k-NN/tree/binary-ivf) against [opensearch-project:feature/binary-format](https://github.com/opensearch-project/k-NN/tree/feature/binary-format) in order to verify on CI test on https://github.com/opensearch-project/k-NN/pull/1784 (you could check this PR for my code change as of now). That's the reason you can see some other code changes on this PR now. CC: @heemin32